client/macro_processor.py

```diff
@@ -184,7 +203,13 @@ def __parse_macros_config_file(self) -> Dict[str, Dict[str, str]]:
                 File name supports wildcard, e.g., with "*.sql", the method will apply the macro map to all the files with
                 extension of ".sql".
         """
-        assert isfile(self.yaml_file_path), "Can't find a file at \"%s\"." % self.yaml_file_path
+        # Try to read the yaml file from either the root dir or the root/client dir.
+        if not isfile(self.yaml_file_path):
+            maybe_file_path = os.path.join("client", self.yaml_file_path)
+            if not isfile(maybe_file_path):
+                raise ValueError("Can't find a file at \"%s\"." % self.yaml_file_path)
+            else:
+                self.yaml_file_path = maybe_file_path
```

Comment:
I'm a bit iffy on this happening down here. This should have happened at a higher level.

Reply:
Done. Now all the args and file-path level validations happen at the very beginning stage, via a validate_args() method.